### PR TITLE
BoardRunner/*: Make more robust and reliable

### LINKED ISSUE #24: Canceling a build results in resume to freed node warning
Currently its possible to "stop" a build by clicking the cross on the sketch window (which removes the entire window)
while this is intended it means that any suspended functions (e.g. functions that yield and are waiting for results) wont be able to return if the sketch window is removed as the calling node is freed.

not critical but would be nice if this didnt happen

### PR DESCRIPTION
**Pull request by RuthgerD**
fixes #24 and generally greatly improves ui <-> runner sync, and possible life time issues.
* A singleton has been introduced to store all the board runners to guarantee some lifetime
* SketchOwner is completely removed
* BoardRunner now has a signal for each state change, including a new "building" state, as thats done asynchonously
* BoardRunner will now remember to free itself when asked but a build is still going
* BoardRunner can now be reset and reused safely, also bringing an auto reconfigure feature
* BoardView and UartSlurper are now valid as long as they exist as the native view / bufs are replaced in place
* ControlPane refactored to make fully use of new state signals, UI wont change unless the board runner actually changed
* ControlPane now manages the vehicle (Still havent found the right place for a dedicated vehicle controller yet, but it works for now)
* Basically all methods where something might fail will return a GDResult with appropriate message

Only thing im not entirely sure about is the ergonomics around the board view invalidation / validation, so how attachments deal with that will change likely

### CODE REVIEW

**Comment by RuthgerD** [comment]
drafted as I havent looked over the diff to see what needs cleanup, but I need to sleep :)

**Comment by RuthgerD** [comment]
After hour+ of testing I havent found any regressions as of yet so merging :)
